drivers/wifi/siwx91x/siwx91x_wifi.c

```diff
@@ -625,6 +649,56 @@ static int siwx91x_stats(const struct device *dev, struct net_stats_wifi *stats)
 }
 #endif
 
+static int siwx91x_mode(const struct device *dev, struct wifi_mode_info *mode)
+{
+	struct siwx91x_dev *sidev = dev->data;
+	sl_wifi_interface_t interface;
+	sl_status_t status;
+	int ret;
+
+	sl_wifi_device_configuration_t network_config = {
+		.boot_option = LOAD_NWP_FW,
+		.band = SL_SI91X_WIFI_BAND_2_4GHZ,
+		.region_code = DEFAULT_REGION,
+		.boot_config = {
+			.tcp_ip_feature_bit_map = (SL_SI91X_TCP_IP_FEAT_DHCPV4_SERVER |
+						  SL_SI91X_TCP_IP_FEAT_EXTENSION_VALID |
+						  SL_SI91X_TCP_IP_FEAT_BYPASS),
+			.custom_feature_bit_map = SL_SI91X_CUSTOM_FEAT_EXTENSION_VALID,
+			.feature_bit_map = SL_SI91X_FEAT_SECURITY_OPEN,
+			.coex_mode = SL_SI91X_WLAN_ONLY_MODE,
```

Comment:
How do you deal with BLE?

I feel this function is in fact shared with Bluetooth and event with Flash driver. Probably it should be located in `soc/silabs/silabs_siwx91x/siwg917/nwp_init.c`.